File: php/tests/RateLimiterTest.php
<?php

declare(strict_types=1);

namespace SecureKit\Tests;

use PHPUnit\Framework\TestCase;
use SecureKit\MemoryRateLimiterStore;
use SecureKit\RateLimiter;

final class RateLimiterTest extends TestCase
{
    public function testAllowsUnderLimitAndBlocksOver(): void
    {
        $rl = new RateLimiter(new MemoryRateLimiterStore(), 3, 60);
        $this->assertTrue($rl->allow('client-1'));
        $this->assertTrue($rl->allow('client-1'));
        $this->assertTrue($rl->allow('client-1'));
        $this->assertFalse($rl->allow('client-1'));
    }

    public function testTracksKeysIndependently(): void
    {
        $rl = new RateLimiter(new MemoryRateLimiterStore(), 1, 60);
        $this->assertTrue($rl->allow('client-A'));
        $this->assertTrue($rl->allow('client-B'));
    }
}

File: java/src/test/java/io/github/securekit/RateLimiterTest.java
package io.github.securekit;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class RateLimiterTest {

    @Test
    void allowsUnderLimitAndBlocksOver() {
        RateLimiter rl = new RateLimiter(new MemoryRateLimiterStore(), 3, 60_000L);
        assertTrue(rl.allow("client-1"));
        assertTrue(rl.allow("client-1"));
        assertTrue(rl.allow("client-1"));
        assertFalse(rl.allow("client-1"));
    }

    @Test
    void tracksKeysIndependently() {
        RateLimiter rl = new RateLimiter(new MemoryRateLimiterStore(), 1, 60_000L);
        assertTrue(rl.allow("client-A"));
        assertTrue(rl.allow("client-B"));
    }
}
